### PR TITLE
[4.0] Popup Messages/Alerts

### PR DESCRIPTION
**Pull request by brianteeman**
In J4 we currently have two types of alert. One uses a custom element and is a "popup" the other is a regular div or "inline". This PR is to improve the accessibility of the popup messages/alerts.

Currently we make use of color which conveys no meaning for a color blind user. Accessibility golden rule is to never rely on just one visual indicator so this PR adds an icon. This also makes the popup alerts consisten with the inline alerts updated #25116

Finally the title of the popup used an H4. @zwiastunsw requested here https://github.com/joomla-projects/custom-elements/issues/99#issuecomment-423668924 that this was not good for accessibility as the headings may not be nested correctly. So I have changed the H4 to a class=alert-heading (I have not customised the style of that class as it would be a waste of time as the admin template would change it) It can always be styled later

This is based on the discussions joomla-projects/custom-elements#99

Easiest way to test would be to select multiple menu items (including the default/home menu item) and try to set them to unpublished.  It should now look like below
image

Related PR #25116

### CODE REVIEW

File: layouts/joomla/system/message.php [hunk 1]
```diff
@@ -36,11 +48,13 @@
...
-								<div><?php echo $msg; ?></div>
-							<?php endforeach; ?>
+						<div class="alert-heading">
+							<span class="<?php echo $icon[$type]; ?>" aria-hidden="true"></span>
+							<?php echo Text::_($type); ?>
+						</div>
+						<?php foreach ($msgs as $msg) : ?>
+							<div><?php echo $msg; ?></div>
+						<?php endforeach; ?>
 						</div>
```
Comment (Quy) [inline layouts/joomla/system/message.php]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ae8d731b05bac69122b8f5cc233fcfd7f74605df<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25135">issues.joomla.org/tracker/joomla-cms/25135</a>.</sub>

**Comment by wilsonge** [comment]
> Accessibility golden rule is to never rely on just one visual indicator so this PR adds an icon

Is the header with the string "Error"/"Message" etc not enough of a visual indicator?

EDIT: Genuine question - after re-reading comes across a bit snarky

**Comment by brianteeman** [comment]
@wilsonge Yes the header could be considered enough. However I think this is still a good idea because
1. It has been shown that people will cognitive issues or dyslexia benefit from the use of icons
2. It adds some consistency with the inline images
3. I still hold out hope that we will drop/reduce these popup messages

**Comment by wilsonge** [comment]
Can you elaborate on the inline image consistency? 

So my gut on this is right now is that it clutters the UI. It definitely makes more sense if the alerts weren’t pop ups and I’d happily reconsider this if/when that happens.

**Comment by brianteeman** [comment]
> Can you elaborate on the inline image consistency?

See #25116 

> So my gut on this is right now is that it clutters the UI.

Says the man who likes orange ;)

Seriously though - up to 8% of users are dyslexic and it is established that the use of visuals are useful to reinforce concepts.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 86a52cb053c95253eaecb2fb51524254fec194c9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25135">issues.joomla.org/tracker/joomla-cms/25135</a>.</sub>

**Comment by ghost** [comment]
@Quy can you please retest?

**Comment by richard67** [comment]
@franz-wohlkoenig Only commit after @Quy 's test was a clean update from base branch without any changes. So it could be set to RTC.

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by dgrammatiko** [comment]
@brianteeman @wilsonge this is wrong!!! You're coupling the alerts system to another dependency (font awesome) and worst part you don't even load that dependency correctly, eg in the layout BUT expecting that the template **will always load font awesome**.

These kinds of code and the underlying assumptions are the blocking elements that make Joomla a really bad option for any front end developer.

The annoying part here is that @C-Lodder already provided the correct code for this part here: https://github.com/joomla-projects/custom-elements/issues/99#issuecomment-423287233
so a pretty straight forward question from me: **WHY ARE YOU CONSTANTLY IGNORING FRONT END DEVELOPERS?**

**Comment by brianteeman** [comment]
Thank you for you constructive comment. It's not correct at all but hey ho I guess you are always right

**Comment by dgrammatiko** [comment]
I will ignore the part about I'm always right and ask you a very simple question **why it's not correct at all?**

- Do you or not coupling to Font awesome **without even loading that asset in that JLayout**?
- Do you expect that **all the Joomla created sites** will **load** font awesome?

Address my honest concerns here and please stop bullying people...

**Comment by brianteeman** [comment]
This is in the admin

**Comment by dgrammatiko** [comment]
@brianteeman you must be joking right?
This file:
https://github.com/joomla/joomla-cms/pull/25135/files#diff-539fd950161fba0becf74d57599e5d39
is used in both front end **and** back end. Same goes for all the fields. That was one of the main reasons I was trying to decouple all fields from Bootstrap AND Font Awesome in order to provide a neutral base for **any** front end framework...

**Comment by wilsonge** [comment]
@dgrammatiko in this case is right - it is shared between frontend and backend (for example save the frontend global config param). Is there a reason charlie's comment on that PR didn't work - because bearing in mind it is used in the frontend - it's probably a better solution

**Comment by richard67** [comment]
I have not tested this item.<br /><br />Resetting test result to not tested until discussion solved. Will test again then.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25135">issues.joomla.org/tracker/joomla-cms/25135</a>.</sub>

**Comment by brianteeman** [comment]
It can be done that way. I didn't because it can't be used the same way for
the inline messages and I assumed it was better to use the same method.
Feel free to submit a pr

**Comment by richard67** [comment]
Who shall do the PR? If nobody else wants, I will try, but I want to avoid to try it while someone else does it at the same time. Credits should go to @C-Lodder anyway.

**Comment by Quy** [comment]
@richard67 Please go for it.

**Comment by richard67** [comment]
@Quy Ok, am working on it.

**Comment by richard67** [comment]
@Quy Hmm, seems the changes which @C-Lodder suggests have to be done in the custom elements repository and not here, or am I wrong?

**Comment by richard67** [comment]
@brianteeman I see there is already an issue for that in the custom elements repo: [https://github.com/joomla-projects/custom-elements/issues/99](https://github.com/joomla-projects/custom-elements/issues/99), is that right? Or am I wrong and it is not related?

**Comment by Quy** [comment]
Yes in the CE repo.

**Comment by richard67** [comment]
Not sure if I get it: Should the same changes as @C-Lodder proposed for the custom elements repo also be done here in file layouts/joomla/system/message.php ?

**Comment by Quy** [comment]
My understanding is that changes in CE will be ported here.

**Comment by richard67** [comment]
Hmm, then it may take some time. Have to prepare for first time contribution to the custom elements.

**Pull request closed without merging**